x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/rule_definition_section.tsx

```diff
@@ -89,16 +89,21 @@ export const Filters = ({
   index,
   'data-test-subj': dataTestSubj,
 }: FiltersProps) => {
-  const flattenedFilters = mapAndFlattenFilters(filters);
-
   const defaultIndexPattern = useDefaultIndexPattern();
-
-  const { indexPattern } = useRuleIndexPattern({
-    dataSourceType: dataViewId ? DataSourceType.DataView : DataSourceType.IndexPatterns,
-    index: index ?? defaultIndexPattern,
-    dataViewId,
-  });
-
+  const useDataViewParams = dataViewId
+    ? { dataViewId }
+    : { indexPatterns: index ?? defaultIndexPattern };
+  const { dataView } = useDataView(useDataViewParams);
+  const isEsql = filters.some((filter) => filter?.query?.language === 'esql');
+  const searchBarFilters = useMemo(() => {
+    if (!index || isDataView(index) || isEsql) {
+      return filters;
+    }
```

Comment:
Can we add some explanation here as to why each of this conditions means that we simply return the unchanged `filters`? 

I specially don't get the the logic for the `isEsql` variable, are we checking if any of the filters are of type ESQL? To my knowledge, filters cannot only be Lucene or KQL. Am I'm missing something?

Reply:
This part was borrowed from another piece of code, suggested in the team, by @nikita.indik, namely the query_bar component. Please see https://github.com/elastic/kibana/blob/b880a1d1ffa2efe9028263c12a0085f31f667b16/x-pack/plugins/security_solution/public/common/components/query_bar/index.tsx#L143